File: benchmarks/fibonacci/fibonacci.go
// fibonacci_benchmark.go
// Go implementation of Fibonacci sequence benchmark
package main

import "fmt"

// Configuration
const N1 = 10
const N2 = 20
const N3 = 30
const N4 = 35

// Markers matching Lamina version
const HEADER_MARKER = 123456789
const FOOTER_MARKER = 987654321

func fibonacci_iterative(n int) int64 {
	if n == 0 {
		return 0
	}
	if n == 1 {
		return 1
	}

	var a int64 = 0
	var b int64 = 1
	for i := 2; i <= n; i++ {
		temp := a + b
		a = b
		b = temp
	}
	return b
}

func main() {
	// Print header marker
	fmt.Println(HEADER_MARKER)

	// Compute and print fibonacci numbers
	fmt.Println(fibonacci_iterative(N1))
	fmt.Println(fibonacci_iterative(N2))
	fmt.Println(fibonacci_iterative(N3))
	fmt.Println(fibonacci_iterative(N4))

	// Print footer marker
	fmt.Println(FOOTER_MARKER)
}

File: benchmarks/fibonacci/fibonacci.rs
// fibonacci_benchmark.rs
// Rust implementation of Fibonacci sequence benchmark

// Configuration
const N1: i32 = 10;
const N2: i32 = 20;
const N3: i32 = 30;
const N4: i32 = 35;

// Markers matching Lamina version
const HEADER_MARKER: i64 = 123456789;
const FOOTER_MARKER: i64 = 987654321;

fn fibonacci_iterative(n: i32) -> i64 {
    if n == 0 {
        return 0;
    }
    if n == 1 {
        return 1;
    }

    let mut a: i64 = 0;
    let mut b: i64 = 1;
    for _ in 2..=n {
        let temp = a + b;
        a = b;
        b = temp;
    }
    b
}

fn main() {
    // Print header marker
    println!("{}", HEADER_MARKER);

    // Compute and print fibonacci numbers
    println!("{}", fibonacci_iterative(N1));
    println!("{}", fibonacci_iterative(N2));
    println!("{}", fibonacci_iterative(N3));
    println!("{}", fibonacci_iterative(N4));

    // Print footer marker
    println!("{}", FOOTER_MARKER);
}
